test/e2e/app/main.go

```diff
@@ -102,51 +110,74 @@ func startNode(cfg *Config) error {
 		return err
 	}
 
-	home := os.Getenv("TMHOME")
-	if home == "" {
-		return errors.New("TMHOME not set")
+	tmcfg, err := setupConfigAndLogger()
```

Comment:
> we replaced the logger variable here 

Where do you mean? In the node constructor?

Reply:
Hehe, exactly. What's currently happening is that we have a global variable `logger` which is updated inside `setupConfigAndLogger()` and then passed to `node.NewNode()`. This isn't very discoverable. It's somewhat better if `setupConfigAndLogger()` returns a new logger, and we update the global `logger` variable here in `startNode` where it's being used.

In general, global variables are a code smell, but loggers are one of the very few exceptions where they sometimes make sense.

Reply:
We're still updating the global `logger` inside `setupNode()`. Create a new `logger` variable instead of reusing the existing global one in `setupNode()`, and update the global in `setupNode()`.